src/image.jl

```diff
@@ -179,6 +179,27 @@ function write(f::FITS, data::Array{T};
     nothing
 end
 
+"""
+    write(hdu::ImageHDU, data::Array)
+
+Write data to an existing image HDU.
+"""
+function write(hdu::ImageHDU, data::Array{T}) where T
+    fits_assert_open(hdu.fitsfile)
+
+    # Ensure sizes are equal
+    hdu_size = fits_get_img_size(hdu.fitsfile)
+    data_size = collect(size(data))
+
+    if hdu_size != data_size
+        error("size of HDU $(hdu_size) not equal to size of data $(data_size).")
+    end
+
+    fits_movabs_hdu(hdu.fitsfile, hdu.ext)
```

Comment:
This (`fits_movabs_hdu`) actually has to come right after the `fits_assert_open` call. The `hdu.fitsfile` C object internally keeps track of which HDU it is on (the "current" HDU). It could be on *any* HDU at the start of this function. So we need to move to the right HDU before checking the image sizes.

Reply:
I'm so dumb :)